### ISSUE TITLE
position elements of legend horizontally

### ISSUE THREAD

**Issue by mo0om** [vega/vega#943]
How do I position each elements of a legend horizontally? I've already tried what is on[ this example](https://github.com/vega/vega/issues/201) but it doesn't seem to work with the current version of Vega. 
Thanks

**Comment by jheer** [comment]
Vega does not (currently) have "native" support for horizontally-oriented legends. However, similar to the linked example, you can override the defaults for custom placement.

One issue here is that the linked example was written for Vega 2. With Vega 3, the custom axis/legend encoding syntax has changed slightly. Instead of `properties` the encoding block should be under an `encode` property, and each subgroup now takes `enter`, `update`, and/or `exit` sets. For more, see the [legend documentation](https://vega.github.io/vega/docs/legends/#custom).

If for some reason this still doesn't work, please feel free to report back and re-open this issue.

**Comment by mo0om** [comment]
I'm close but I can't get it to work. here is the code...
![image](https://user-images.githubusercontent.com/1723490/28728858-42b83184-7398-11e7-95fe-c2886deefa31.png)

```
{
  "$schema": "https://vega.github.io/schema/vega/v3.0.json",
  "width": 500,
  "height": 300,
  "padding": 20,
  "signals": [
    { "name": "cellSize", "value": 100 }
  ],
  "scales": [
    {
      "name": "x",
      "type": "time",
      "range": "width",
      "domain": {"data": "stats", "field": "x"}
    },
    {
      "name": "y",
      "type": "linear",
      "range": "height",
      "nice": true,
      "zero": true,
      "domain": {"data": "stats", "field": "y"}
    },

    {
      "name": "colorlegend", 
      "type": "ordinal",
      "domain": {"data": "stats", "field": "graph"}, 
      "range": "category"
    },
     {
      "name": "legend-series-x",
      "type": "ordinal",
      "domain": {"data": "stats", "field": "graph"}, 
      "range": [0, 100]
    }
  ],
   "legends": [
    {
      "fill": "colorlegend",
      "orient": "top",
      "encode": {
          "symbols": {
            "update": {
              "x": {"scale": "legend-series-x", "field": "graph"},
              "y": [{"value": 20}],
              "fillOpacity": {"value": 0.5},
              "stroke": {"value": "transparent"}
            }
          },
          "labels": {
            "update": {
              "x": {"scale": "legend-series-x", "field": "graph", "offset": 15},
              "y": [{"value": 20}]
            }
          }
        
      }
    }
  ],
  "marks": [
    {
      "type": "group",
      "from": {
        "facet": {
          "name": "series",
          "data": "stats",
          "groupby": "graph"
        }
      },
      "marks": [
        {
          "type": "line",
          "from": {
            "data": "series"
            },
          "encode": {
            "enter": {
              "x": {"scale": "x", "field": "x"},
              "y": {"scale": "y", "field": "y"},
              "stroke": {"scale": "colorlegend", "field": "graph"},
              "strokeWidth": {"value": 3},
              "strokeOpacity": {"value": 0.7}
            },
            "update": {
              "fillOpacity": {"value": 1}
            },
            "hover": {
              "fillOpacity": {"value": 0.5}
            }
          }
        }
      ]
    }
  ],
 "axes": [
    { "orient": "bottom", "scale": "x", "domain": false, "format": "%H:%M", "grid": true },
    { "orient": "left", "scale": "y", "title": "%", "grid": true }
  ],
  "data": [
    {
      "name": "stats",
      "values":[
            {
              "graph": "graph A",
              "x": 1500492300000,
              "y": 63.87579568625058
            },
            {
              "graph": "graph A",
              "x": 1500492420000,
              "y": 140.14065294117647
            },
            {
              "graph": "graph A",
              "x": 1500492540000,
              "y": 72.7588843137255
            },
             {
              "graph": "graph B",
              "x": 1500492300000,
              "y": 63.056507451016294
            },
            {
              "graph": "graph B",
              "x": 1500492420000,
              "y": 200.446230588187426
            },
            {
              "graph": "graph B",
              "x": 1500492540000,
              "y": 72.30749882357728
            }
          ]
    }
  ]
}
```

**Comment by jheer** [comment]
Almost there! All that remains is to replace `"field": "graph"` with `"field": "value"` in your custom legend encoders. Underneath the hood, Vega generates new data sets that back legends and axes. These data items include `value` and `label` attributes.

**Comment by mo0om** [comment]
yes!!! It works. Thanks Jeffrey.

**Comment by ericsoco** [comment]
FWIW, horizontal legend orientation is now available in 4.0. (`"direction": "horizontal"`).
https://github.com/vega/vega/issues/1005
